### PR TITLE
Use malloc/free for LRUHandle instead of new[]/delete[]

### PR DESCRIPTION
**Pull request by ltamasi**
Summary:
It's unsafe to call `malloc_usable_size` with an address not returned by a function from the `malloc` family (see #10798). The patch switches from using `new[]` / `delete[]` for `LRUHandle` to `malloc` / `free`.

Test Plan:
`make check`

### CODE REVIEW

File: cache/lru_cache.cc [hunk 1]
```diff
@@ -560,7 +560,7 @@ LRUHandle* LRUCacheShard::Lookup(const Slice& key, uint32_t hash,
                                  is_in_sec_cache);
     if (secondary_handle != nullptr) {
       e = reinterpret_cast<LRUHandle*>(
-          new char[sizeof(LRUHandle) - 1 + key.size()]);
+          port::cacheline_aligned_alloc(sizeof(LRUHandle) - 1 + key.size()));
```
Comment (pdillinger) [inline cache/lru_cache.cc:563]: This is a material space usage increase for unknown / supposed speed increase. With 16 byte key, LRUHandle is currently size ~88 bytes, which jemalloc rounds up to 96 bytes. You are rounding up to 128. Successful Lookup() always accesses the beginning (`value`) and end (`key_data`) of the LRUHandle, so it guaranteed to access two cache lines regardless of alignment efforts. Release() non-last reference only accesses `refs` so only accesses one cache line (assuming minimum alignment guaranteed by malloc). Release() last reference (LRU_Insert) modifies `next` and `m_flags` which will be in different cache lines (unless the LRUHandle starts 40 - 54 bytes offset from a cache line, which will not happen in jemalloc unless we optimize LRUHandle size down one class, and then only 1 chance in 4).

Do you have a rationale?

I'm also not sure *how* guaranteed malloc_usable_size is on the result of posix_memalign. Some searching suggests "probably."

**Review by pdillinger — CHANGES_REQUESTED** [review]

File: cache/lru_cache.cc (hunk 1 above)
Comment (ltamasi) [reply cache/lru_cache.cc:563]: Thanks for the analysis - fair enough. Will switch to plain old `malloc`.

File: cache/lru_cache.cc [hunk 2]
```diff
@@ -560,7 +560,7 @@ LRUHandle* LRUCacheShard::Lookup(const Slice& key, uint32_t hash,
                                  is_in_sec_cache);
     if (secondary_handle != nullptr) {
       e = reinterpret_cast<LRUHandle*>(
```
Comment (pdillinger) [inline cache/lru_cache.cc:562]: `static_cast` can be used for better hygiene (cleaner looking; detect possible bugs under future modification)

**Review by pdillinger — COMMENTED** [review]
LGTM

**Review by pdillinger — APPROVED** [review]

**Pull request closed without merging**